src/common/codeine/ConsulRegistrator.java

```diff
@@ -35,12 +36,8 @@ public boolean consulExists() {
     public void register(String name, int port) {
         log.info("Will register " + name + " with port " + port);
         AgentClient agentClient = client.agentClient();
-        Registration service = ImmutableRegistration.builder()
-            .id(name)
-            .name(name)
-            .port(port)
-            .check(Registration.RegCheck.tcp("localhost:" + port, 10, 5))
-            .build();
+        Registration service = ImmutableRegistration.builder().id(name).name(name).port(port)
+            .check(Registration.RegCheck.http("localhost:" + port + "/" + Constants.HEALTH_CONTEXT, 60, 5)).build();
```

Comment:
All of the other components in our environment are using the ip for registration and health checks

Reply:
i am not sure i need to use the ip address for health checks registration. The peer is registering with the IP already